File: src/org/mtransit/android/commons/KeyboardUtils.java
package org.mtransit.android.commons;

import android.app.Activity;
import android.content.Context;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

public final class KeyboardUtils implements MTLog.Loggable {

	private static final String TAG = KeyboardUtils.class.getSimpleName();

	@Override
	public String getLogTag() {
		return TAG;
	}

	public static void showKeyboard(Activity activity, View view) {
		if (activity == null || view == null) {
			return;
		}
		InputMethodManager imm = (InputMethodManager) activity.getSystemService(Context.INPUT_METHOD_SERVICE);
		if (imm != null) {
			imm.showSoftInput(view, InputMethodManager.SHOW_IMPLICIT);
		}
	}

	public static void hideKeyboard(Activity activity, View view) {
		if (activity == null || view == null) {
			return;
		}
		InputMethodManager imm = (InputMethodManager) activity.getSystemService(Context.INPUT_METHOD_SERVICE);
		if (imm != null) {
			imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
		}
	}

}

File: src/main/java/org/mtransit/android/commons/KeyboardUtils.kt
package org.mtransit.android.commons

import android.app.Activity
import android.view.View
import android.view.inputmethod.InputMethodManager
import androidx.core.content.getSystemService

@Suppress("unused")
class KeyboardUtils : MTLog.Loggable {

    override fun getLogTag() = LOG_TAG

    companion object {
        private val LOG_TAG: String = KeyboardUtils::class.java.simpleName

        @JvmStatic
        fun showKeyboard(activity: Activity?, view: View?) {
            view ?: return
            val imm = activity?.getSystemService<InputMethodManager>() ?: return
            imm.showSoftInput(view, InputMethodManager.SHOW_IMPLICIT)
        }

        @JvmStatic
        fun hideKeyboard(activity: Activity?, view: View?) {
            view ?: return
            val imm = activity?.getSystemService<InputMethodManager>() ?: return
            imm.hideSoftInputFromWindow(view.windowToken, 0)
        }
    }
}